global-script.lua

```diff
@@ -2896,20 +2931,120 @@ function upCastPosSizRot(oPos,size,rot,dist,multi,tags)
+        end
+    end
+    if not initialized then
+        obj.setVar("initialized", true)
+        -- Energy Cost (button index 0)
+        obj.createButton({
+            label="", click_function="nullFunc",
+            position={0,2.24,-11.2}, rotation={0,180,0}, height=0, width=0,
+            font_color={1,1,1}, font_size=500
+        })
+        -- Sit Here (button index 1)
+        obj.createButton({
+            label="", click_function="onClickedSitHere", function_owner=Global,
+            position={ 7,2.24,-24.7}, rotation={0,180,0}, height=0, width=0,
+            font_color={0,0,0}, font_size=500,
+            tooltip="Moves your current player color to be located here.  The color currently seated here will be moved to your current location.  Spirit panels and other cards will be relocated if applicable.",
+        })
+        -- Change Color (button index 2)
+        obj.createButton({
+            label="", click_function="onClickedChangeColor", function_owner=Global,
+            position={-7,2.24,-24.7}, rotation={0,180,0}, height=0, width=0,
+            font_color={0,0,0}, font_size=500,
+            tooltip="Change to be this color, updating all of your presence and reminder tokens accordingly.  The player that is this color will be changed to be yours.  Your seating position will not change.",
+        })
+        -- Play Spirit (button index 3)
+        obj.createButton({
+            label="", click_function="onClickedPlaySpirit", function_owner=Global,
+            position={0,2.24,-24.7}, rotation={0,180,0}, height=0, width=0,
+            font_color={0,0,0}, font_size=500,
+            tooltip="Switch to play the spirit that is here, changing your player color accordingly.  Only available for spirits without a seated player.  Intended for multi-handed solo games.",
         })
+        -- Other buttons to follow/be fixed later.
+    end
+
+    if selected then
+        obj.editButton({index=0, label="Energy Cost: ?"})
+    else
+        obj.editButton({index=0, label=""})
+    end
+
+    if showPlayerButtons then
+        local bg = Color[color]
+        local fg
+        if (bg.r*0.30 + bg.g*0.59 + bg.b*0.11) > 0.50 then
+            fg = {0,0,0}
+        else
+            fg = {1,1,1}
+        end
+        obj.editButton({index=1, label="Sit Here", height=800, width=3300})
+        obj.editButton({index=2, label="Pick " .. color, height=800, width=3300, color=bg, font_color=fg})
+    else
+        obj.editButton({index=1, label="", height=0, width=0})
+        obj.editButton({index=2, label="", height=0, width=0})
+    end
+
+    if Player[color].seated or (not selected and not showAllMultihandedButtons) then
+        obj.editButton({index=3, label="", height=0, width=0})
+    else
+        obj.editButton({index=3, label="Play Spirit", height=800, width=3300})
+    end
+
+    local label = ""
+    if selected then
+        label = "?"
+    end
+    for _,bag in pairs(params.elementBags) do
```

Comment:
what's going on here?

Reply:
Rather than selectively add/delete buttons (which also causes their indices to shift around), buttons always exist.  By changing them to have an empty label and a height/width of zero, they effectively are invisible.  This technique is already used for the spirit panels for the power progression and aspect decks.

The counters for elements are subject to this showing and hiding as well.  When initially showing them, I opted to set the text to "?" instead of "0", which is immediately overwritten by countItems().  If for some reason something breaks, this will cause energy costs to display as ? rather than 0 which provides a better indication to what actually broke then briefly showing 0s.